haystack/components/joiners/branch.py

```diff
@@ -11,6 +11,10 @@
 logger = logging.getLogger(__name__)
 
 
+class JoinedBranchesDict(TypedDict):
+    value: Any
```

Comment:
I'm unsure about this. 

Using this type hinting for Components that have a "static" output it's fine for me, though I don't see much value in doing the same for those with dynamic output. The only pro is that you know that the `dict` will contain a `value` key in this case. Though I'm unsure about the type, it can be `Any` type but it's still a known type after the Component is created. 🤔

Reply:
Okay, got you. Is there any downside in adding the typing or would it be an option to have a type that defines that the output can just be known after creation ? 🤔

Reply:
(We would just need to have any type. Since we try to suggest components to the user that might be compatible, this component can't be suggested. If it would be dict, we could just say "might". This happens before configuring the component)

Reply:
I don't really see the need to define the type after creation, every I/O definition is accessible through `__haystack_input__` and `__haystack_output__` in any Component, you can already get it from there. 

Adding the type for the Components with "static" I/O is fine but it crumbles with those like the `ConditionalRouter`. You can't know anything about its output until you create an instance, neither name nor type.

It also seems earily similar to one of the first experiments we made to defined Components I/O, and in the end we discarded cause of the above limitations.